userlist.ts

```diff
@@ -0,0 +1,303 @@
+import { UserItem, UserHtmlElements } from "./types";
+import { data } from "./data";
+import { User } from "./user";
+import { callElements } from "./app";
+import { validate } from "./validate";
+
+let button: HTMLElement = document.getElementById("changeColor");
+let table: HTMLElement = document.getElementById("tablebody");
+
+export class UserList {
+  "users": UserItem[] = [];
+
+  constructor(init?: UserItem[]) {
+    Object.assign(this.users, init);
+  }
+
+  loadData = () => {
+    let items: UserItem[] = [...this.users] as UserItem[];
+
+    let itemsToDisplay: string[] = this.makeLayout(items);
+
+    table.innerHTML = itemsToDisplay.join("");
+
+    button.innerText = "Refresh data";
+    button.onclick = this.loadData;
+
+    //adding event to add new user
+    let submitnewuserbtn: HTMLElement = document.getElementById(
```

Comment:
htmlbuttonelement

Reply:
Working on this!
Changing it to HTMLButtonElement is giving some error.